insights_content_template_renderer/js_executor.py

```diff
@@ -0,0 +1,166 @@
+"""
+JavaScript execution pool for running JS code in an isolated process.
+
+This module provides a process pool for executing JavaScript code using PythonMonkey.
+Running JS in the separate process avoids the PythonMonkey FastAPI recursion bug
+(see https://github.com/Distributive-Network/PythonMonkey/issues/490).
+"""
+
+import logging
+import multiprocessing as mp
+
+
+log = logging.getLogger(__name__)
+
+
+def _eval_js_worker_task(js_code, data):
+    """
+    Execute JavaScript in a worker process.
+    This function is called by pool workers and returns the result directly.
+
+    :param js_code: JavaScript code to execute
+    :param data: Data to pass to the JavaScript function
+    :return: Tuple of (status, result) where status is 'success' or 'error'
+    """
+    try:
+        import pythonmonkey as pm
+        func = pm.eval(js_code)
+        result = func(data)
+        # Convert to native Python string to avoid pickling issues
+        # PythonMonkey returns JS strings that can't be pickled
+        result_str = str(result) if result is not None else ''
+        return ('success', result_str)
+    except Exception as e:
+        import traceback
+        return ('error', f"{str(e)}\n{traceback.format_exc()}")
+
+
+class JsExecutor:
+    """
+    Manages a process pool for executing JavaScript code in isolated processes.
+
+    This class provides a singleton-style process pool that is lazily initialized
+    on first use. Each uvicorn worker gets its own pool with a single worker process.
+    """
+
+    def __init__(self):
+        """Initialize the JsExecutor with no process pool (lazy initialization)."""
+        self._process_pool = None
+        self._pool_lock = mp.Lock()
+        self._timeout = 5  # Default timeout in seconds
+
+    def get_pool(self):
+        """
+        Get or create the process pool for JavaScript execution.
+        Uses lazy initialization to avoid creating processes during module import.
+
+        Uses a single worker process since each uvicorn worker has its own pool.
+
+        :return: The process pool instance
+        """
+        if self._process_pool is None:
+            with self._pool_lock:
+                # Double-check pattern to avoid race conditions
+                if self._process_pool is None:
+                    log.info("Initializing JavaScript worker process")
+
+                    # Use spawn method to avoid inheriting FastAPI context
+                    ctx = mp.get_context('spawn')
+                    self._process_pool = ctx.Pool(processes=1)
```

Comment:
If the pool will be always size 1, why do we need an executor pool at all? It can be handled using a simple `Process` object pointing to the function that you want to run?